client/src/main/java/io/pravega/client/stream/impl/PositionImpl.java

```diff
@@ -118,13 +118,14 @@ protected PositionBuilder newBuilder() {
 
         @Override
         protected byte getWriteVersion() {
-            return 0;
+            return 1;
```

Comment:
This will break backward compatibility. Old code does not know how to deal with version 1.

Do this using a revision.

Reply:
This bug prevents serialization with the older version of the client for -1L.

I used a new version number since the serializing via `RevisionDataOutput#writeCompactSignedLong` (new change) cannot be deserialized via `RevisionedDataOutput#readCompactLong` (old change).

I have also added a test to verify if serialization via version 0 can be deserialized via version 1.

Reply:
Done, I have updated the PR to ensure it is compatible both ways.